lib/src/analytics.dart

```diff
@@ -27,6 +29,18 @@ class Analytics {
       forceReset: forceReset ?? false,
     );
     initializer.run();
+
+    // Initialize the config handler class and check if the
+    // tool message and version have been updated from what
+    // is in the current file; if there is a new message version
+    // make the necessary updates
+    // ignore: unused_local_variable TODO: remove after increment method implemented
+    bool messagePrinted = false;
+    if (!_configHandler.parsedTools.containsKey(tool)) {
+      _configHandler.addTool(tool: tool);
+      stdout.writeln(toolsMessage);
```

Comment:
I don't think you can send this to stdout. Consider if the client is running in machine mode, and supposed to output valid JSON to stdout.

In hixie's original doc, the analytics API exposed two public methods that clients could call: `shouldShowMessage()` and `didShowMessage()`

Reply:
Ah yes, good point, thanks for pointing that out, we can have the client itself decide how to show the message appropriately